### PR TITLE
Add prettier

### PR DESCRIPTION
**Pull request by sharplet**

### CODE REVIEW

File: TemplateProject/.prettierrc [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "semi": false,
+  "singleQuote": false,
```
Comment (gfontenot) [inline TemplateProject/.prettierrc]: I _think_ this is the default? I could be wrong.

File: TemplateProject/.vimrc [hunk 2]
```diff
@@ -0,0 +1 @@
+let g:ale_javascript_prettier_options = '--config .prettierrc'
```
Comment (gfontenot) [inline TemplateProject/.vimrc]: Does this not get set by default? We might let users set this themselves locally. I don't feel strongly about this though.

Comment (sharplet) [reply TemplateProject/.vimrc]: Unfortunately it doesn't. Given that there's a project-specific style, I don't think there's any harm in including this for ALE users. I think it's currently up to the tools to know about and support prettier config files.

File: TemplateProject/.prettierrc (hunk 1 above)
Comment (sharplet) [reply TemplateProject/.prettierrc]: You are correct!

File: TemplateProject/.prettierrc [hunk 3]
```diff
@@ -0,0 +1,4 @@
+{
+  "semi": false,
```
Comment (jakecraige) [inline TemplateProject/.prettierrc]: 🔥 🔥 🔥

**Pull request merged**